src/hello_client.erl

```diff
@@ -165,12 +166,16 @@ handle_call(terminate, _From, State) ->
 %% @hidden
 handle_info({?INCOMING_MSG, Message}, State) ->
     incoming_message(Message, State);
+handle_info(?PING, State = #client_state{waiting_for_pong = true, keep_alive_interval = KeepAliveInterval}) ->
+    ?LOG_ERROR("There is no PONG answer on PING for ~p msec. Client will be restarted.", [KeepAliveInterval]),
```

Comment:
to 1) yes i know, but the whole client state is also dismissed =)
to 2) yes that would be nice. I know this sounds a bit over the top but knowing the time of a connection loss can be crucial in troubleshooting. If you just restart the client transport you could save such a timestamp in the client state and restart the transport without loosing it. 

What do you exactly want to do with these metrics? Just log them? For the timeout we should define a certain 'KeepAliveTimeout' Interval or whatever to be clear about when the client reestablishes the connection.